redaxo/src/addons/cronjob/lib/command/run.php

```diff
@@ -35,4 +47,48 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $io->success('Cronjobs skipped.');
         return 0;
     }
+
+    private function executeSingleJob(SymfonyStyle $io, $id)
+    {
+        $manager = rex_cronjob_manager_sql::factory();
+
+        if (null === $id) {
+            $jobs = rex_sql::factory()->getArray('
+                SELECT id, name
+                FROM ' . REX_CRONJOB_TABLE . '
+                WHERE environment LIKE "%|script|%"
+                ORDER BY id
+            ');
+            $jobs = array_column($jobs, 'name', 'id');
+
+            $question = new ChoiceQuestion('Which cronjob should be executed?', $jobs);
+            $question->setValidator(function ($selected) use ($jobs) {
```

Comment:
Wenn man die choices von außen vorgibt brauch man den validator glaub ich nicht

Reply:
Default ist [dieser Validator](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Question/ChoiceQuestion.php#L131-L182).
Der Validator liefert gleichzeitig den letztlich zurückgegebenen Wert, also der, der unten aus `askQuestion` rauskommt.
Der Default-Validator liefert da nicht den Key, sondern den Wert, in unserem Fall den Cronjob-Namen. Der ist aber nicht eindeutig.
Daher habe ich keine andere Möglichkeit gesehen, als einen eigenen Validator zu schreiben, der den Key stattdessen liefert.